examples/medplum-chart-demo/src/bots/core/charting-utils.ts

```diff
@@ -0,0 +1,374 @@
+import { getReferenceString } from '@medplum/core';
+import {
+  Bundle,
+  BundleEntry,
+  ClinicalImpression,
+  CodeableConcept,
+  Condition,
+  Encounter,
+  Observation,
+  ObservationComponent,
+  Patient,
+  Practitioner,
+  Quantity,
+  QuestionnaireResponse,
+  QuestionnaireResponseItemAnswer,
+  Reference,
+  Resource,
+} from '@medplum/fhirtypes';
+import { randomUUID } from 'crypto';
+
+export interface BloodPressure {
+  systolic?: Quantity;
+  diastolic?: Quantity;
+}
+
+/**
+ * This function takes data about observations and returns an array of Observation resources.
+ *
+ * @param observationData - An easily parseable object containing data for various observation resources
+ * @param codes - A map of observation types to their coding
+ * @param encounter - The encounter the observations are derived from
+ * @param user - The user creating the observations
+ * @param response - The QuestionnaireResponse that the observations are being created from
+ * @returns An array of Observation resources.
+ */
+export function createObservations(
+  observationData: Record<string, QuestionnaireResponseItemAnswer>,
+  codes: Record<string, CodeableConcept>,
+  encounter: Encounter,
+  user: Practitioner,
+  response: QuestionnaireResponse
+): Observation[] {
+  const observations: Observation[] = [];
+
+  for (const [key, value] of Object.entries(observationData)) {
+    if (!value || key === 'date') {
```

Comment:
since date is handled specially, why don't you just break it out into it's own input argmuent, rather than sticking it into `observationData`?